src/redux/store.tsx

```diff
@@ -34,7 +35,7 @@ const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
 export const store = createStore(
     rootReducer,
     compose(
-        applyMiddleware(socketMiddleware(socketService.get())),
+        applyMiddleware(initialStartUpMiddleware(socketService.get()), chartMiddleware(socketService.get()), searchMiddleware(socketService.get())),
```

Comment:
You're eagerly establishing a WebSocket at module load time by getting it here, rather than passing the SocketService through and creating it during bootstrapping.